client/src/Components/RecordingsBrowser/RecordingsBrowser.jsx

```diff
@@ -102,20 +104,24 @@ export default function RecordingsBrowser(props) {
 
   // Load in the connection info based on repoId
   useEffect(() => {
-    window.fftData = {}; // prevents the previously viewed spectrogram from showing up breifly at the start of another
-
     // This will happen when someone is linked directly to an azure repo
     if (location.search && !connection.accountName) {
       console.log('Updating connection info and fetching recordings list!');
       const accountName = searchParams.get('accountName');
       const containerName = searchParams.get('containerName');
       const sasToken = searchParams.get('sasToken');
-      updateConnectionAccountName(accountName);
-      updateConnectionContainerName(containerName);
-      updateConnectionSasToken(sasToken);
-      fetchRecordingsList({ accountName: accountName, containerName: containerName, sasToken: sasToken });
+      dispatch(updateConnectionAccountName(accountName));
+      dispatch(updateConnectionContainerName(containerName));
+      dispatch(updateConnectionSasToken(sasToken));
```

Comment:
This makes me spicy

Reply:
It is the current way that is working we hope to change this soon